src/components/implementation/tests/unit_torrent/torrent_test.c

```diff
@@ -6,116 +6,223 @@
+	char *str4 = " Diablo 1 and Torchlight are the best! Also Borderlands.";
+	char *g = cbuf_alloc(strlen(str4), &cb4);
+	if (!g) { printc("UNIT TEST FAILED: split2 failed %c\n", *g); return; }
+	memcpy(g, str4, strlen(str4));
+	snprintf(val, 8, "%d", 23);
+	rt = twmeta(cos_spd_id(), t1, "offset", strlen("offset"),
+		 val, strlen(val));
+	c = twritep(cos_spd_id(), t1, cb4, 0, strlen(str4));
+	
+	// Close the file so we can read from it
 	trelease(cos_spd_id(), t1);
-	printc("UNIT TEST PASSED: split->release\n");
 
-	t1 = tsplit(cos_spd_id(), td_root, params2, strlen(params2), TOR_ALL, evt1);
+
+
+
+
+	/* End of writing stage */
+	/* Going to test reading next */
+
+	// now test read
+	a = 0; b = 0; c = 0;
+	t1 = tsplit(cos_spd_id(), td_root, params1, strlen(params1), TOR_ALL, evt1);
 	if (t1 < 1) {
 		printc("UNIT TEST FAILED: split2 failed %d\n", t1); return;
 	}
-	t2 = tsplit(cos_spd_id(), t1, params1, strlen(params1), TOR_ALL, evt2);
-	if (t2 < 1) {
-		printc("UNIT TEST FAILED: split3 failed %d\n", t2); return;
-	}
 
-	ret1 = twrite_pack(cos_spd_id(), t1, data1, strlen(data1));
-	ret2 = twrite_pack(cos_spd_id(), t2, data2, strlen(data2));
-	printv("write %d & %d, ret %d & %d\n", strlen(data1), strlen(data2), ret1, ret2);
+	printc("\n\ncalling read\n");
+	c = treadp(cos_spd_id(), t1, &a, &b);
+	char *buf = cbuf2buf(c, 6);
+	printc("Expected:[%s] Actual[%s]\n", buf, d);
 
+	printc("UNIT TEST Unit tests for torrents - specifically treadp, twritep... are done\n");
 	trelease(cos_spd_id(), t1);
-	trelease(cos_spd_id(), t2);
-	printc("UNIT TEST PASSED: 2 split -> 2 write -> 2 release\n");
-
-	t1 = tsplit(cos_spd_id(), td_root, params2, strlen(params2), TOR_ALL, evt1);
-	t2 = tsplit(cos_spd_id(), t1, params1, strlen(params1), TOR_ALL, evt2);
-	if (t1 < 1 || t2 < 1) {
-		printc("UNIT TEST FAILED: later splits failed\n");
-		return;
+
+	unsigned int i, j;
+	int str_length = 0;
+	unsigned long long t; unsigned long rnd;
+	td_t t3;
+	long evt2 = 0;
+	char *params2 = "boo";
+	t3 = tsplit(cos_spd_id(), td_root, params2, strlen(params1), TOR_ALL, evt2);
+	if (t3 < 1) {
+		printc("UNIT TEST FAILED: split2 failed %c\n", t3); return;
 	}
 	
-	ret1 = tread_pack(cos_spd_id(), t1, buffer, 1023);
-	if (ret1 > 0) buffer[ret1] = '\0';
-	printv("read %d (%d): %s (%s)\n", ret1, strlen(data1), buffer, data1);
-	assert(!strcmp(buffer, data1));
-	assert(ret1 == strlen(data1));
-	buffer[0] = '\0';
-
-	ret1 = tread_pack(cos_spd_id(), t2, buffer, 1023);
-	if (ret1 > 0) buffer[ret1] = '\0';
-	assert(!strcmp(buffer, data2));
-	assert(ret1 == strlen(data2));
-	printv("read %d: %s\n", ret1, buffer);
-	buffer[0] = '\0';
-
-	trelease(cos_spd_id(), t1);
-	trelease(cos_spd_id(), t2);
+	for (i = 0; i < 100; i++) {
+		
+		
+		rdtscll(t);
+		rnd = (int) (t & 127);
+		int rnd_val;
+		
+		if (rnd > 0 && rnd < 42) {
+			rnd_val = 0;
+		}
+		else if (rnd >= 42 && rnd < 84) {
+			rnd_val = 1;
+		}
+		else
+		{
+			rnd_val = 2;
+		}
+
+		int offset;
+		switch (rnd_val) {
+			case 0:
```

Comment:
I'm not convinced that you understand `switch` from this.  Please make sure you do.`